tests/test_sim_trait.py

```diff
@@ -0,0 +1,775 @@
+        assert len(sim_result["EffectSize"]) == num_causal
+
+    def test_output_dim_gamma(self):
+        model = tstrait.trait_model(distribution="gamma", shape=1, scale=1)
+        random_seed = 1
+        num_ind = 10
+        num_causal = 5
+        alpha = 1
+        ts = msprime.sim_ancestry(
+            num_ind, sequence_length=100_000, random_seed=random_seed
+        )
+        ts = msprime.sim_mutations(ts, rate=0.01, random_seed=random_seed)
+        sim_result = tstrait.sim_traits(
+            ts=ts,
+            num_causal=num_causal,
+            model=model,
+            alpha=alpha,
+            random_seed=random_seed,
+        )
+        assert sim_result.shape[0] == num_causal
+        assert sim_result.shape[1] == 3
+        assert len(sim_result["SiteID"]) == num_causal
+        assert len(sim_result["CausalState"]) == num_causal
+        assert len(sim_result["EffectSize"]) == num_causal
+
+
+class Test_sim_phenotype_input:
+    @pytest.mark.parametrize("ts", [0, "a", [1, 1]])
+    def test_ts(self, ts):
+        model = tstrait.trait_model(distribution="normal", mean=0, var=1)
+        with pytest.raises(TypeError, match="Input must be a tree sequence data"):
+            tstrait.sim_traits(ts=ts, num_causal=3, model=model, alpha=1, random_seed=1)
+
+    @pytest.mark.parametrize("num_causal", ["1", "a", [1, 1]])
+    def test_num_causal(self, num_causal):
+        ts = msprime.sim_ancestry(2, sequence_length=100_000, random_seed=1)
+        ts = msprime.sim_mutations(ts, rate=0.01, random_seed=1)
+        model = tstrait.trait_model(distribution="normal", mean=0, var=1)
+        with pytest.raises(
+            TypeError, match="Number of causal sites must be an integer"
+        ):
+            tstrait.sim_traits(
+                ts=ts, num_causal=num_causal, model=model, alpha=1, random_seed=1
+            )
+
+    @pytest.mark.parametrize("num_causal", [-1, 1.8, -1.5, 0])
+    def test_num_causal_positive(self, num_causal):
+        ts = msprime.sim_ancestry(2, sequence_length=100_000, random_seed=1)
+        ts = msprime.sim_mutations(ts, rate=0.01, random_seed=1)
+        model = tstrait.trait_model(distribution="normal", mean=0, var=1)
+        with pytest.raises(
+            ValueError, match="Number of causal sites must be a positive integer"
+        ):
+            tstrait.sim_traits(
+                ts=ts, num_causal=num_causal, model=model, alpha=1, random_seed=1
+            )
+
+    @pytest.mark.parametrize("model", ["normal", 1, [None, 1, "a"]])
+    def test_model(self, model):
+        ts = msprime.sim_ancestry(5, sequence_length=100_000, random_seed=2)
+        ts = msprime.sim_mutations(ts, rate=0.01, random_seed=2)
+        with pytest.raises(
+            TypeError, match="Trait model must be an instance of TraitModel"
+        ):
+            tstrait.sim_traits(ts=ts, num_causal=1, model=model, alpha=1, random_seed=1)
+
+    @pytest.mark.parametrize("num_ind", [1, 2, np.array([5])[0]])
```

Comment:
There's really no point in taking the cross product of these parameters - they don't interact and all you're testing is whether you raise a ValueError. It's just a waste of time and electricity!